## 1. Setup and Installation

First, let's install and import the required libraries.

In [ ]:
# Install required packages
#!pip install pystac-client pandas requests matplotlib seaborn geopandas folium -q

In [ ]:
# Import libraries
import json
import pandas as pd
from pystac_client import Client
import pystac
from datetime import datetime, timedelta
import matplotlib.pyplot as plt
import seaborn as sns
from typing import List, Dict, Any
import warnings

warnings.filterwarnings('ignore')

# Set display options for better readability
pd.set_option('display.max_rows', 20)
pd.set_option('display.max_columns', None)
pd.set_option('display.max_colwidth', 50)
pd.set_option('display.width', None)

# Plotting style
sns.set_style('whitegrid')
plt.rcParams['figure.figsize'] = (12, 6)

print("✅ All libraries imported successfully!")

✅ All libraries imported successfully!


---

## 2. Connecting to the Montandon STAC API

Let's establish a connection to the API and explore its basic structure.

In [ ]:
# Connect to Montandon STAC API using pystac_client
STAC_API_URL = "https://montandon-eoapi-stage.ifrc.org/stac"
client = Client.open(STAC_API_URL)

# Get API information
api_info = client.get_root().to_dict()

print("🌐 Connected to Montandon STAC API")
print(f"API Title: {api_info.get('title', 'N/A')}")
print(f"API Description: {api_info.get('description', 'N/A')}")
print(f"\nAPI Endpoint: {STAC_API_URL}")

🌐 Connected to Montandon STAC API
API Title: stac-fastapi
API Description: stac-fastapi

API Endpoint: https://montandon-eoapi-stage.ifrc.org/stac


### Understanding API Conformance

The API conforms to various STAC specifications. Let's check what capabilities are available:

In [ ]:
# Check API conformance and capabilities
try:
    root = client.get_root().to_dict()
    conformance = root.get("conformsTo", [])

    print("📋 API Conformance Classes:")
    for idx, conf in enumerate(conformance, 1):
        conf_name = conf.split("/")[-1] if "/" in conf else conf
        print(f"  {idx}. {conf_name}")
except Exception as exc:
    print("Could not retrieve conformance information.")
    print(f"Reason: {exc}")
    print("You can continue using the API for data queries.")

📋 API Conformance Classes:
  1. basic-cql2
  2. cql2-json
  3. cql2-text
  4. simple-query
  5. core
  6. geojson
  7. oas30
  8. features-filter
  9. filter
  10. collection-search
  11. collection-search#fields
  12. collection-search#filter
  13. collection-search#free-text
  14. collection-search#query
  15. collection-search#sort
  16. item-search#filter
  17. collections
  18. transaction
  19. core
  20. item-search
  21. item-search#fields
  22. item-search#query
  23. item-search#sort
  24. ogcapi-features
  25. ogcapi-features#fields
  26. ogcapi-features#query
  27. ogcapi-features#sort
  28. transaction


In [ ]:
# Define helper function using pystac_client
from typing import Optional

def search_stac(
    collections: Optional[List[str]] = None,
    limit: int = 100,
    bbox: Optional[List[float]] = None,
    datetime_range: Optional[str] = None,
    query: Optional[Dict[str, Any]] = None,
    sortby: Optional[List[str]] = None,
    filter_body: Optional[Dict[str, Any]] = None,
    filter_lang: Optional[str] = None,
) -> list:
    """
    Search STAC API using pystac_client.
    
    Parameters:
    -----------
    collections : list of str
        Collection IDs to search
    limit : int
        Maximum number of results to return
    bbox : list of float
        Bounding box [min_lon, min_lat, max_lon, max_lat]
    datetime_range : str
        ISO 8601 datetime range "start/end"
    query : dict
        Query filters
    sortby : list
        Sort fields
    filter_body : dict
        CQL2 filter body
    filter_lang : str
        Filter language (cql2-json)
    
    Returns:
    --------
    list of pystac.Item
        All matching items
    """
    search = client.search(
        collections=collections,
        max_items=limit,
        bbox=bbox,
        datetime=datetime_range,
        query=query,
        sortby=sortby,
        filter=filter_body,
        filter_lang=filter_lang,
    )
    return list(search.get_items())


print("✅ PySTAC helper function initialized")

✅ PySTAC helper function initialized


---

## 3. Exploring Available Collections

Collections in the Montandon API are organized by data source and type:

- **Events**: Main disaster event records
- **Hazards**: Specific hazard information (earthquakes, floods, etc.)
- **Impacts**: Impact assessments (casualties, economic losses, etc.)

Let's explore all available collections:

In [ ]:
# Get all collections
collections = list(client.get_collections())

print(f"📚 Total Collections Available: {len(collections)}\n")

# Create a DataFrame for better visualization
collection_data = []
for collection in collections:
    collection_data.append({
        'Collection ID': collection.id,
        'Title': collection.title if hasattr(collection, 'title') else 'N/A',
        'Description': collection.description[:80] + '...' if collection.description and len(collection.description) > 80 else collection.description,
        'License': collection.license if hasattr(collection, 'license') else 'N/A'
    })

collections_df = pd.DataFrame(collection_data)
collections_df

📚 Total Collections Available: 29



Collection ID                                             Title  \
0    desinventar-events                         DesInventar Mapped Events   
1   desinventar-impacts                               DesInventar Impacts   
2          emdat-events                              EM-DAT Source Events   
3         emdat-hazards                             EM-DAT Source Hazards   
4         emdat-impacts                             EM-DAT Source Impacts   
5          gdacs-events                               GDACS Source Events   
6         gdacs-hazards                                     GDACS Hazards   
7         gdacs-impacts                                     GDACS Impacts   
8            gfd-events                                 GFD Source Events   
9           gfd-hazards                                GFD Source Hazards   
10          gfd-impacts                                       GFD Impacts   
11         glide-events                               GLIDE Source Events   
12        glide-hazards                              GLIDE Source Hazards   
13       ibtracs-events                             IBTrACS Source Events   
14      ibtracs-hazards                                   IBTrACS Hazards   
15     idmc-gidd-events                           IDMC GIDD Source Events   
16    idmc-gidd-impacts                                 IDMC GIDD Impacts   
17      idmc-idu-events  IDMC Internal Displacement Updates (IDU) Impacts   
18     idmc-idu-impacts  IDMC Internal Displacement Updates (IDU) Impacts   
19     ifrcevent-events                                IFRC Source Events   
20    ifrcevent-hazards                               IFRC Source Hazards   
21    ifrcevent-impacts                               IFRC Source Impacts   
22           pdc-events                                 PDC Source Events   
23          pdc-hazards                                PDC Source Hazards   
24          pdc-impacts                                       PDC Impacts   
25     reference-events                                  Reference Events   
26          usgs-events                                       USGS Events   
27         usgs-hazards                                      USGS Hazards   
28         usgs-impacts                                      USGS Impacts   

                                                                            Description  \
0   Events mapped from the DesInventar disaster loss database. DesInventar is a conc...   
1   Impact records mapped from the DesInventar disaster loss database. DesInventar i...   
2   Global Disaster Events from the Emergency Events Database (EM-DAT). EM-DAT is a ...   
3   Hazard records from the Emergency Events Database (EM-DAT). EM-DAT is a global d...   
4   Impact records from the Emergency Events Database (EM-DAT). EM-DAT is a global d...   
5   Events from the Global Disaster Alert and Coordination System (GDACS). GDACS is ...   
6   Hazard records from the Global Disaster Alert and Coordination System (GDACS). G...   
7   Impact records from the Global Disaster Alert and Coordination System (GDACS). G...   
8   Events from the Global Flood Database (GFD). GFD combines several years of flood...   
9   Hazard records from the Global Flood Database (GFD). GFD combines several years ...   
10  Impact records from the Global Flood Database (GFD). GFD combines several years ...   
11  Events from the GLobal IDEntifier Number (GLIDE) system. GLIDE is a globally com...   
12  Hazard records from the GLobal IDEntifier Number (GLIDE) system. GLIDE is a glob...   
13  International Best Track Archive for Climate Stewardship (IBTrACS) - A global da...   
14  A collection of tropical cyclone hazards from the International Best Track Archi...   
15  Events from the Internal Displacement Monitoring Centre (IDMC) Global Internal D...   
16  Impact records from the Internal Displacement Monitoring Centre (IDMC) Global In...   
17  Events from the Internal Displacement Monitoring Centre (

### Categorizing Collections by Type

Let's organize collections by their type (events, hazards, impacts):

In [ ]:
# Categorize collections
events_collections = [c for c in collections if '-events' in c.id]
hazards_collections = [c for c in collections if '-hazards' in c.id]
impacts_collections = [c for c in collections if '-impacts' in c.id]

print("📊 Collections by Type:\n")
print(f"🔹 Events Collections ({len(events_collections)}):")
for col in events_collections:
    print(f"   • {col.id}")

print(f"\n🔹 Hazards Collections ({len(hazards_collections)}):")
for col in hazards_collections:
    print(f"   • {col.id}")

print(f"\n🔹 Impacts Collections ({len(impacts_collections)}):")
for col in impacts_collections:
    print(f"   • {col.id}")

📊 Collections by Type:

🔹 Events Collections (12):
   • desinventar-events
   • emdat-events
   • gdacs-events
   • gfd-events
   • glide-events
   • ibtracs-events
   • idmc-gidd-events
   • idmc-idu-events
   • ifrcevent-events
   • pdc-events
   • reference-events
   • usgs-events

🔹 Hazards Collections (8):
   • emdat-hazards
   • gdacs-hazards
   • gfd-hazards
   • glide-hazards
   • ibtracs-hazards
   • ifrcevent-hazards
   • pdc-hazards
   • usgs-hazards

🔹 Impacts Collections (9):
   • desinventar-impacts
   • emdat-impacts
   • gdacs-impacts
   • gfd-impacts
   • idmc-gidd-impacts
   • idmc-idu-impacts
   • ifrcevent-impacts
   • pdc-impacts
   • usgs-impacts


### Categorizing by Data Source

In [ ]:
# Extract data sources
sources = {}
for col in collections:
    source = col.id.split('-')[0]
    if source not in sources:
        sources[source] = []
    sources[source].append(col.id)

print("Collections by Data Source:\n")
for source, cols in sorted(sources.items()):
    print(f"{source.upper()} ({len(cols)} collections):")
    for col in sorted(cols):
        print(f"   • {col}")
    print()

Collections by Data Source:

DESINVENTAR (2 collections):
   • desinventar-events
   • desinventar-impacts

EMDAT (3 collections):
   • emdat-events
   • emdat-hazards
   • emdat-impacts

GDACS (3 collections):
   • gdacs-events
   • gdacs-hazards
   • gdacs-impacts

GFD (3 collections):
   • gfd-events
   • gfd-hazards
   • gfd-impacts

GLIDE (2 collections):
   • glide-events
   • glide-hazards

IBTRACS (2 collections):
   • ibtracs-events
   • ibtracs-hazards

IDMC (4 collections):
   • idmc-gidd-events
   • idmc-gidd-impacts
   • idmc-idu-events
   • idmc-idu-impacts

IFRCEVENT (3 collections):
   • ifrcevent-events
   • ifrcevent-hazards
   • ifrcevent-impacts

PDC (3 collections):
   • pdc-events
   • pdc-hazards
   • pdc-impacts

REFERENCE (1 collections):
   • reference-events

USGS (3 collections):
   • usgs-events
   • usgs-hazards
   • usgs-impacts



### Deep Dive: Examining a Specific Collection

Let's examine the GDACS hazards collection in detail:

In [ ]:
# Get a specific collection
gdacs_hazards = client.get_collection('gdacs-hazards')

print("📖 Collection Details: gdacs-hazards\n")
print(f"Title: {gdacs_hazards.title}")
print(f"Description: {gdacs_hazards.description}")
print(f"License: {gdacs_hazards.license}")

# Spatial extent
if hasattr(gdacs_hazards, 'extent') and gdacs_hazards.extent:
    spatial = gdacs_hazards.extent.spatial
    temporal = gdacs_hazards.extent.temporal
    
    print(f"\n🌍 Spatial Extent:")
    print(f"   Bounding Box: {spatial.bboxes}")
    
    print(f"\n📅 Temporal Extent:")
    print(f"   Intervals: {temporal.intervals}")

# Additional properties
if hasattr(gdacs_hazards, 'extra_fields'):
    print(f"\n🔧 Additional Properties:")
    for key, value in gdacs_hazards.extra_fields.items():
        print(f"   {key}: {value}")

📖 Collection Details: gdacs-hazards

Title: GDACS Hazards
Description: Hazard records from the Global Disaster Alert and Coordination System (GDACS). GDACS is a cooperation framework between the United Nations, the European Commission and disaster managers worldwide to improve alerts, information exchange and coordination in the first phase after major sudden-onset disasters. It provides detailed hazard information including affected areas, alert levels, and severity scores for earthquakes, tsunamis, floods, tropical cyclones, volcanic eruptions, and wildfires. Each hazard includes specific information based on the hazard type and uses specialized models for assessment. More information on the GDACS mapping in Monty can be found in the [GDACS Hazard Source Mappings](https://github.com/IFRCGo/monty-stac-extension/tree/main/model/sources/GDACS#hazard-item).
License: MIT

🌍 Spatial Extent:
   Bounding Box: [[-180, -90, 180, 90]]

📅 Temporal Extent:
   Intervals: [[datetime.datetime(2000, 

---

## 4. Basic Data Retrieval - Simple Queries

Now let's start retrieving actual data! We'll begin with simple queries.

### Example 1: Retrieve 10 Items from GDACS Hazards

In [ ]:
# Simple search with limit using pystac_client
search = client.search(collections=['gdacs-hazards'], max_items=10)
items = list(search.get_items())

print(f"✅ Retrieved {len(items)} items from gdacs-hazards collection\n")

# Display first item details
if items:
    first_item = items[0]
    print("📄 First Item Details:")
    print(f"ID: {first_item.id}")
    print(f"Collection: {first_item.collection_id}")
    print(f"Geometry Type: {first_item.geometry['type'] if first_item.geometry else 'None'}")
    print(f"\nProperties:")
    for key, value in list(first_item.properties.items())[:10]:  # Show first 10 properties
        print(f"  • {key}: {value}")

✅ Retrieved 10 items from gdacs-hazards collection

📄 First Item Details:
ID: gdacs-hazard-1510840-1672502
Collection: gdacs-hazards
Geometry Type: Point

Properties:
  • roles: ['source', 'hazard']
  • title: Earthquake in Japan
  • source: NEIC
  • datetime: 2025-11-18T08:21:21Z
  • keywords: ['Earthquake', 'JPN', 'Seismic', 'Geological']
  • description: Green M 4.8 Earthquake in Japan at: 18 Nov 2025 08:21:21.
  • end_datetime: 2025-11-18T08:21:21Z
  • monty:etl_id: 6d565ef1-ff16-46fa-b265-4b6f6364d3bf
  • severitydata: {'severity': 4.8, 'severitytext': 'Magnitude 4.8M, Depth:45.026km', 'severityunit': 'M'}
  • monty:corr_id: 20251118-JPN-GH0101-1-GCDB


### Example 2: Using Direct API Requests

Sometimes you might want to use direct HTTP requests instead of the Python client:

In [ ]:
# Direct API request using pystac_client
collection = client.get_collection('gdacs-hazards')
search = client.search(collections=['gdacs-hazards'], max_items=5)
items = list(search.get_items())

print(f"📡 PySTAC Client Response")
print(f"\n📊 Results:")
print(f"  • Number of Items: {len(items)}")
print(f"  • Collection: {collection.id}")

# Display feature IDs
if items:
    print(f"\n🔖 Item IDs:")
    for item in items:
        print(f"  • {item.id}")

📡 PySTAC Client Response

📊 Results:
  • Number of Items: 5
  • Collection: gdacs-hazards

🔖 Item IDs:
  • gdacs-hazard-1510840-1672502
  • gdacs-hazard-1510833-1672493
  • gdacs-hazard-1510834-1672495
  • gdacs-hazard-1001235-1
  • gdacs-hazard-1510796-1672457


---

## 5. Advanced Filtering Techniques

Now let's explore more sophisticated filtering options.

### 5.1 Temporal Filtering with DateTime

Filter events by date range:

In [ ]:
# Define date range - last 90 days
end_date = datetime.now()
start_date = end_date - timedelta(days=90)

# Format as ISO 8601
datetime_range = f"{start_date.strftime('%Y-%m-%dT%H:%M:%SZ')}/{end_date.strftime('%Y-%m-%dT%H:%M:%SZ')}"

print(f"📅 Searching for events between:")
print(f"   Start: {start_date.strftime('%Y-%m-%d')}")
print(f"   End: {end_date.strftime('%Y-%m-%d')}\n")

# Search with datetime filter using pystac_client
items = search_stac(
    collections=['gdacs-events'],
    datetime_range=datetime_range,
    limit=20
)

print(f"✅ Found {len(items)} events in the last 90 days")

# Display event details
if items:
    print("\n📋 Recent Events:")
    for i, item in enumerate(items[:5], 1):
        event_date = item.properties.get('datetime', 'N/A')
        event_name = item.properties.get('title', item.id)
        print(f"  {i}. {event_name} - {event_date}")

📅 Searching for events between:
   Start: 2025-08-22
   End: 2025-11-20

✅ Found 20 events in the last 90 days

📋 Recent Events:
  1. Earthquake in Japan - 2025-11-18T08:21:21Z
  2. Earthquake in Argentina - 2025-11-18T07:11:32Z
  3. Earthquake in United States - 2025-11-18T07:08:29Z
  4. Tropical Cyclone FIVE-25 - 2025-11-18T06:00:00Z
  5. Earthquake in Japan - 2025-11-18T02:05:52Z
✅ Found 20 events in the last 90 days

📋 Recent Events:
  1. Earthquake in Japan - 2025-11-18T08:21:21Z
  2. Earthquake in Argentina - 2025-11-18T07:11:32Z
  3. Earthquake in United States - 2025-11-18T07:08:29Z
  4. Tropical Cyclone FIVE-25 - 2025-11-18T06:00:00Z
  5. Earthquake in Japan - 2025-11-18T02:05:52Z


### 5.2 Spatial Filtering with Bounding Box (bbox)

Filter events by geographic area. Let's search for events in Southeast Asia:

In [ ]:
# Define bounding box for Southeast Asia [min_lon, min_lat, max_lon, max_lat]
southeast_asia_bbox = [95.0, -10.0, 141.0, 28.0]

print("🗺️ Searching for events in Southeast Asia")
print(f"   Bounding Box: {southeast_asia_bbox}\n")

# Search with bbox using pystac_client
items = search_stac(
    collections=['gdacs-events', 'emdat-events', 'usgs-events'],
    bbox=southeast_asia_bbox,
    limit=30
)

print(f"✅ Found {len(items)} events in Southeast Asia")

# Analyze by collection
if items:
    collection_counts = {}
    for item in items:
        col = item.collection_id
        collection_counts[col] = collection_counts.get(col, 0) + 1
    
    print("\n📊 Events by Collection:")
    for col, count in sorted(collection_counts.items()):
        print(f"  • {col}: {count} events")

🗺️ Searching for events in Southeast Asia
   Bounding Box: [95.0, -10.0, 141.0, 28.0]

✅ Found 30 events in Southeast Asia

📊 Events by Collection:
  • gdacs-events: 17 events
  • usgs-events: 13 events
✅ Found 30 events in Southeast Asia

📊 Events by Collection:
  • gdacs-events: 17 events
  • usgs-events: 13 events


### 5.3 Combining Multiple Filters

Let's combine temporal and spatial filters:

In [ ]:
# Search for recent events in Europe
europe_bbox = [-10.0, 35.0, 40.0, 71.0]
recent_datetime = "2024-01-01T00:00:00Z/2024-12-31T23:59:59Z"

print("🔍 Searching for 2024 events in Europe\n")

items = search_stac(
    collections=['gdacs-events', 'emdat-events'],
    bbox=europe_bbox,
    datetime_range=recent_datetime,
    limit=50
)

print(f"✅ Found {len(items)} events in Europe during 2024")

# Display sample
if items:
    print("\n📋 Sample Events:")
    for i, item in enumerate(items[:3], 1):
        print(f"\n  {i}. ID: {item.id}")
        print(f"     Collection: {item.collection_id}")
        print(f"     Title: {item.properties.get('title', 'N/A')}")
        print(f"     Date: {item.properties.get('datetime', 'N/A')}")
        print(f"     Country: {item.properties.get('monty:country_codes', 'N/A')}")

🔍 Searching for 2024 events in Europe

✅ Found 50 events in Europe during 2024

📋 Sample Events:

  1. ID: emdat-event-2024-0960-TUN
     Collection: emdat-events
     Title: Water in Tunisia of December 2024
     Date: 2024-12-31T00:00:00Z
     Country: ['TUN']

  2. ID: emdat-event-2024-0954-ITA
     Collection: emdat-events
     Title: Water in Italy of December 2024
     Date: 2024-12-31T00:00:00Z
     Country: ['ITA']

  3. ID: emdat-event-2024-0949-ESP
     Collection: emdat-events
     Title: Water in Spain of December 2024
     Date: 2024-12-26T00:00:00Z
     Country: ['ESP']
✅ Found 50 events in Europe during 2024

📋 Sample Events:

  1. ID: emdat-event-2024-0960-TUN
     Collection: emdat-events
     Title: Water in Tunisia of December 2024
     Date: 2024-12-31T00:00:00Z
     Country: ['TUN']

  2. ID: emdat-event-2024-0954-ITA
     Collection: emdat-events
     Title: Water in Italy of December 2024
     Date: 2024-12-31T00:00:00Z
     Country: ['ITA']

  3. ID: emdat-event

### 5.4 Advanced Filtering with CQL2

Common Query Language (CQL2) allows for complex filtering logic. Let's search for flood events in Spain (like the example from the documentation):

### 5.5 Finding Hazards Associated with an Event

Once we have an event, we can find all related hazards using the correlation ID:

In [ ]:

spain_flood_late_oct_filter = {
    "op": "and",
    "args": [
        {
            "op": "a_contains",
            "args": [
                {"property": "monty:country_codes"},
                "ESP"
            ]
        },
        {
            "op": "a_overlaps",
            "args": [
                {"property": "monty:hazard_codes"},
                ["nat-hyd-flo-flo", "FL"]
            ]
        },
        {
            "op": "t_intersects",
            "args": [
                {"property": "datetime"},
                {"interval": ["2024-10-20T00:00:00Z", "2024-11-05T23:59:59Z"]}
            ]
        }
    ]
}

spain_flood_late_oct_events = search_stac(
    collections=["gdacs-events", "emdat-events", "glide-events", "reference-events"],
    filter_body=spain_flood_late_oct_filter,
    filter_lang="cql2-json",
    limit=100
)

print(f"✅ Found {len(spain_flood_late_oct_events)} flood events in Spain (Oct 20 - Nov 5, 2024)\n")

if spain_flood_late_oct_events:
    print(f"📋 All {len(spain_flood_late_oct_events)} Spain Late October/Early November Flood Events:\n")
    for i, event in enumerate(spain_flood_late_oct_events, 1):
        print(f"\n{i}. Event Details:")
        print(f"   ID: {event.id}")
        print(f"   Title: {event.properties.get('title', 'N/A')}")
        print(f"   Date: {event.properties.get('datetime', 'N/A')}")
        print(f"   Collection: {event.collection_id}")
        print(f"   Countries: {event.properties.get('monty:country_codes', 'N/A')}")
        print(f"   Hazard Codes: {event.properties.get('monty:hazard_codes', 'N/A')}")
        print(f"   Correlation ID: {event.properties.get('monty:corr_id', 'N/A')}")
else:
    print("⚠️ No flood events found in Spain during Oct 20 - Nov 5, 2024")

✅ Found 4 flood events in Spain (Oct 20 - Nov 5, 2024)

📋 All 4 Spain Late October/Early November Flood Events:


1. Event Details:
   ID: gdacs-event-1102983
   Title: Flood in Spain
   Date: 2024-10-27T15:00:00Z
   Collection: gdacs-events
   Countries: ['ESP']
   Hazard Codes: ['FL']
   Correlation ID: 20241027-ESP-NAT-HYD-FLO-FLO-1-GCDB

2. Event Details:
   ID: glide-event-FL-2024-000199-ESP
   Title: 
   Date: 2024-10-27T00:00:00Z
   Collection: glide-events
   Countries: ['ESP']
   Hazard Codes: ['nat-hyd-flo-flo', 'FL']
   Correlation ID: 20241027-ESP-NAT-HYD-FLO-FLO-1-GCDB

3. Event Details:
   ID: emdat-event-2024-0796-ESP
   Title: Flood (General) in Spain
   Date: 2024-10-27T00:00:00Z
   Collection: emdat-events
   Countries: ['ESP']
   Hazard Codes: ['nat-hyd-flo-flo']
   Correlation ID: 20241027-ESP-NAT-HYD-FLO-FLO-1-GCDB

4. Event Details:
   ID: gdacs-event-1102955
   Title: Flood in Spain
   Date: 2024-10-11T01:00:00Z
   Collection: gdacs-events
   Countries: ['ESP']
 

In [ ]:
# Get correlation IDs from Spain flood events and find related hazards
if spain_flood_late_oct_events:
    # Collect unique correlation IDs
    correlation_ids = set()
    for event in spain_flood_late_oct_events:
        corr_id = event.properties.get('monty:corr_id')
        if corr_id:
            correlation_ids.add(corr_id)
    
    print(f"🔗 Found {len(correlation_ids)} unique correlation IDs from Spain flood events\n")
    print("="*80)
    
    # Search for hazards for each correlation ID
    all_hazards = []
    hazard_collections_used = ["glide-hazards", "gdacs-hazards", "usgs-hazards", "reference-hazards"]
    
    for idx, corr_id in enumerate(sorted(correlation_ids), 1):
        print(f"\n📍 CORRELATION ID {idx}/{len(correlation_ids)}: {corr_id}")
        print("="*80)
        
        # Search for related hazards using CQL2 filter
        hazard_filter = {
            "op": "=",
            "args": [
                {"property": "monty:corr_id"},
                corr_id
            ]
        }
        
        hazards = search_stac(
            collections=hazard_collections_used,
            filter_body=hazard_filter,
            filter_lang="cql2-json",
            limit=50
        )
        
        print(f"\n✅ Found {len(hazards)} hazard(s) across collections: {', '.join(hazard_collections_used)}\n")
        
        if hazards:
            all_hazards.extend(hazards)
            
            # Group hazards by collection for clearer display
            hazards_by_collection = {}
            for hazard in hazards:
                coll = hazard.collection_id
                if coll not in hazards_by_collection:
                    hazards_by_collection[coll] = []
                hazards_by_collection[coll].append(hazard)
            
            # Display hazards grouped by collection
            for collection_name, coll_hazards in sorted(hazards_by_collection.items()):
                print(f"📦 Collection: {collection_name} ({len(coll_hazards)} hazard{'s' if len(coll_hazards) > 1 else ''})")
                print("-" * 80)
                
                for i, hazard in enumerate(coll_hazards, 1):
                    print(f"\n  {i}. Hazard ID: {hazard.id}")
                    print(f"     Date: {hazard.properties.get('datetime', 'N/A')}")
                    print(f"     Title: {hazard.properties.get('title', 'N/A')}")
                    
                    # Display hazard detail if available
                    hazard_detail = hazard.properties.get('monty:hazard_detail', {})
                    if hazard_detail:
                        severity = hazard_detail.get('severity_value', 'N/A')
                        severity_unit = hazard_detail.get('severity_unit', '')
                        cluster = hazard_detail.get('cluster', 'N/A')
                        print(f"     Severity: {severity} {severity_unit}")
                        print(f"     Cluster: {cluster}")
                    else:
                        print(f"     Severity: N/A")
                        print(f"     Cluster: N/A")
                
                print()
        else:
            print("⚠️ No hazards found for this correlation ID")
    
    print("\n" + "="*80)
    print(f"\n🎯 FINAL SUMMARY")
    print("="*80)
    print(f"Total Correlation IDs: {len(correlation_ids)}")
    print(f"Total Hazards Found: {len(all_hazards)}")
    
    # Summary by collection
    if all_hazards:
        print(f"\n📊 Hazards by Collection:")
        collection_summary = {}
        for hazard in all_hazards:
            coll = hazard.collection_id
            collection_summary[coll] = collection_summary.get(coll, 0) + 1
        
        for coll, count in sorted(collection_summary.items()):
            print(f"  • {coll}: {count} hazard{'s' if count > 1 else ''}")
    
else:
    print("ℹ️ No Spain flood events found to search for hazards")

🔗 Found 2 unique correlation IDs from Spain flood events


📍 CORRELATION ID 1/2: 20241011-ESP-NAT-HYD-FLO-FLO-1-GCDB

✅ Found 1 hazard(s) across collections: glide-hazards, gdacs-hazards, usgs-hazards, reference-hazards

📦 Collection: gdacs-hazards (1 hazard)
--------------------------------------------------------------------------------

  1. Hazard ID: gdacs-hazard-1102955-1
     Date: 2024-10-11T01:00:00Z
     Title: Flood in Spain
     Severity: 0.5 GDACS Severity Score
     Cluster: nat-hyd-flo-flo


📍 CORRELATION ID 2/2: 20241027-ESP-NAT-HYD-FLO-FLO-1-GCDB

✅ Found 3 hazard(s) across collections: glide-hazards, gdacs-hazards, usgs-hazards, reference-hazards

📦 Collection: gdacs-hazards (2 hazards)
--------------------------------------------------------------------------------

  1. Hazard ID: gdacs-hazard-1102983-2
     Date: 2024-10-27T15:00:00Z
     Title: Flood in Spain
     Severity: 0.5 GDACS Severity Score
     Cluster: nat-hyd-flo-flo

  2. Hazard ID: gdacs-hazard-110298

### 5.6 Finding Impacts Associated with Spain Flood Events

Now let's search for impact data related to the Spain flood events:

In [ ]:
from pystac_client import Client
from datetime import datetime

# Connect to Montandon STAC API
client = Client.open("https://montandon-eoapi-stage.ifrc.org/stac")

# Define the filter
filter_dict = {
    "op": "and",
    "args": [
        {
            "op": "a_contains",
            "args": [
                {"property": "monty:country_codes"},
                "ESP"
            ]
        },
        {
            "op": "a_overlaps",
            "args": [
                {"property": "monty:hazard_codes"},
                ["MH0600", "nat-hyd-flo-flo", "FL"]
            ]
        },
        {
            "op": "t_intersects",
            "args": [
                {"property": "datetime"},
                {"interval": ["2024-10-27T00:00:00Z", "2024-10-28T23:59:59Z"]}
            ]
        },
        
    ]
}

# Search for events
search = client.search(
    collections=["glide-events", "gdacs-events", "emdat-events", "reference-events"],
    filter=filter_dict,
    filter_lang="cql2-json",
    limit=100
)

# Retrieve all matching items
events = list(search.items())

print(f"Found {len(events)} related events:")
for event in events:
    print(f"  - {event.id} from {event.collection_id}")
    print(f"    Correlation ID: {event.properties.get('monty:corr_id', 'N/A')}")
    print(f"    Date: {event.properties.get('datetime')}")

Found 4 related events:
  - gdacs-event-1102983 from gdacs-events
    Correlation ID: 20241027-ESP-NAT-HYD-FLO-FLO-1-GCDB
    Date: 2024-10-27T15:00:00Z
  - glide-event-FL-2024-000199-ESP from glide-events
    Correlation ID: 20241027-ESP-NAT-HYD-FLO-FLO-1-GCDB
    Date: 2024-10-27T00:00:00Z
  - emdat-event-2024-0796-ESP from emdat-events
    Correlation ID: 20241027-ESP-NAT-HYD-FLO-FLO-1-GCDB
    Date: 2024-10-27T00:00:00Z
  - gdacs-event-1102955 from gdacs-events
    Correlation ID: 20241011-ESP-NAT-HYD-FLO-FLO-1-GCDB
    Date: 2024-10-11T01:00:00Z


In [ ]:
# Search for impacts related to Spain flood events
if spain_flood_late_oct_events:
    print("🔍 Searching for impacts related to Spain flood events\n")
    print("="*80)
    
    # Define impact filter for Spain floods
    impact_filter = {
        "op": "and",
        "args": [
            {
                "op": "a_contains",
                "args": [
                    {"property": "monty:country_codes"},
                    "ESP"
                ]
            },
            {
                "op": "a_overlaps",
                "args": [
                    {"property": "monty:hazard_codes"},
                    ["nat-hyd-flo-flo", "FL"]
                ]
            },
            {
                "op": "t_intersects",
                "args": [
                    {"property": "datetime"},
                    {"interval": ["2024-10-20T00:00:00Z", "2024-11-15T23:59:59Z"]}
                ]
            }
        ]
    }
    
    # Search across impact collections
    impact_collections = ["gdacs-impacts", "emdat-impacts", "idmc-gidd-impacts", "idmc-idu-impacts"]
    
    impacts = search_stac(
        collections=impact_collections,
        filter_body=impact_filter,
        filter_lang="cql2-json",
        limit=100
    )
    
    print(f"✅ Found {len(impacts)} impact(s) across collections: {', '.join(impact_collections)}\n")

🔍 Searching for impacts related to Spain flood events

✅ Found 46 impact(s) across collections: gdacs-impacts, emdat-impacts, idmc-gidd-impacts, idmc-idu-impacts

✅ Found 46 impact(s) across collections: gdacs-impacts, emdat-impacts, idmc-gidd-impacts, idmc-idu-impacts



In [ ]:
# Diagnostic: Examine the actual structure of impact items
if impacts:
    print("🔍 DIAGNOSTIC: Examining Impact Data Structure\n")
    print("="*80)
    
    # Sample a few impacts from different collections
    sample_impacts = {}
    for impact in impacts[:10]:
        coll = impact.collection_id
        if coll not in sample_impacts:
            sample_impacts[coll] = impact
    
    for collection_name, impact in sample_impacts.items():
        print(f"\n📦 Collection: {collection_name}")
        print(f"Impact ID: {impact.id}")
        print("-" * 80)
        
        # Show all properties
        print("\n📋 All Properties:")
        for key, value in impact.properties.items():
            if isinstance(value, dict):
                print(f"  {key}:")
                for sub_key, sub_value in value.items():
                    print(f"    {sub_key}: {sub_value}")
            else:
                print(f"  {key}: {value}")
        
        # Show monty:impact_detail specifically
        impact_detail = impact.properties.get('monty:impact_detail', {})
        if impact_detail:
            print(f"\n⚡ monty:impact_detail structure:")
            for key, val in impact_detail.items():
                print(f"    {key}: {val}")
        else:
            print(f"\n⚠️ monty:impact_detail is empty or missing")
        
        print("\n" + "="*80)
else:
    print("No impacts to examine")

🔍 DIAGNOSTIC: Examining Impact Data Structure


📦 Collection: idmc-idu-impacts
Impact ID: idmc-idu-impact-28064168623-displaced
--------------------------------------------------------------------------------

📋 All Properties:
  roles: ['source', 'impact']
  title: Spain: Flood - Andalusia (Málaga) - 13/11/2024
  sources: Local Authorities
  datetime: 2024-11-13T00:00:00Z
  location: Alhaurín de la Torre, Malaga, Andalusia, Spain; Álora, Malaga, Andalusia, Spain; Cártama, Malaga, Andalusia, Spain; Málaga, Malaga, Andalusia, Spain; Pizarra, Malaga, Andalusia, Spain; Vélez-Málaga, La Axarquía, Malaga, Andalusia, Spain
  description:  **Spain: 4,210 displacements, 13 November - 14 November**    
 According to local authority, a total of 4210 people were displaced in Álora, Cártama, Alhaurín de la Torre, Pizarra, Málaga, Vélez-Málaga due to flood on 13 November.   
 [Local Authorities - 14 November 2024](https://www.juntadeandalucia.es/presidencia/portavoz/emergencias112/198838/suspension

In [ ]:

if spain_flood_late_oct_events: 
    if impacts:
        # Group impacts by collection
        impacts_by_collection = {}
        for impact in impacts:
            coll = impact.collection_id
            if coll not in impacts_by_collection:
                impacts_by_collection[coll] = []
            impacts_by_collection[coll].append(impact)
        
        # Display impacts grouped by collection
        for collection_name, coll_impacts in sorted(impacts_by_collection.items()):
            print(f"\n📦 Collection: {collection_name} ({len(coll_impacts)} impact{'s' if len(coll_impacts) > 1 else ''})")
            print("-" * 80)
            
            for i, impact in enumerate(coll_impacts, 1):
                print(f"\n  {i}. Impact ID: {impact.id}")
                print(f"     Correlation ID: {impact.properties.get('monty:corr_id', 'N/A')}")
                print(f"     Date: {impact.properties.get('datetime', 'N/A')}")
                print(f"     Title: {impact.properties.get('title', 'N/A')}")
                
                # Display impact details if available
                impact_detail = impact.properties.get('monty:impact_detail', {})
                if impact_detail:
                    print(f"\n     Impact Detail:")
                    print(f"       • Category: {impact_detail.get('category', 'N/A')}")
                    print(f"       • Type: {impact_detail.get('type', 'N/A')}")
                    
                    # Format value with proper number formatting
                    value = impact_detail.get('value', 'N/A')
                    if value != 'N/A' and isinstance(value, (int, float)):
                        print(f"       • Value: {value:,.0f}")
                    else:
                        print(f"       • Value: {value}")
                    
                    print(f"       • Unit: {impact_detail.get('unit', 'N/A')}")
                    print(f"       • Estimate Type: {impact_detail.get('estimate_type', 'N/A')}")
                    
                    # Display description if available
                    description = impact_detail.get('description')
                    if description:
                        print(f"       • Description: {description}")
                else:
                    print(f"     Impact Details: Not available")
            
            print()
        
        # Summary statistics
        print("\n" + "="*80)
        print(f"\n🎯 IMPACTS SUMMARY")
        print("="*80)
        print(f"Total Impacts Found: {len(impacts)}")
        
        # Summary by collection
        print(f"\n📊 Impacts by Collection:")
        for coll, impact_list in sorted(impacts_by_collection.items()):
            count = len(impact_list)
            print(f"  • {coll}: {count} impact{'s' if count > 1 else ''}")
        
        # Calculate aggregated impact statistics by type
        impact_stats = {}
        
        for impact in impacts:
            impact_detail = impact.properties.get('monty:impact_detail', {})
            if impact_detail:
                impact_type = impact_detail.get('type', 'unknown')
                category = impact_detail.get('category', 'unknown')
                value = impact_detail.get('value')
                unit = impact_detail.get('unit', '')
                
                if value and isinstance(value, (int, float)):
                    key = f"{category}_{impact_type}"
                    if key not in impact_stats:
                        impact_stats[key] = {
                            'category': category,
                            'type': impact_type,
                            'unit': unit,
                            'total': 0,
                            'count': 0
                        }
                    impact_stats[key]['total'] += value
                    impact_stats[key]['count'] += 1
        
        # Display aggregated impact statistics
        if impact_stats:
            print(f"\n📈 Aggregated Impact Statistics:")
            print("(Note: Some values may overlap across different data sources)\n")
            
            for key, stats in sorted(impact_stats.items()):
                impact_label = f"{stats['category'].title()} - {stats['type'].replace('_', ' ').title()}"
                avg_value = stats['total'] / stats['count']
                
                print(f"  • {impact_label}:")
                print(f"      Total: {stats['total']:,.0f} {stats['unit']}")
                print(f"      Average: {avg_value:,.0f} {stats['unit']}")
                print(f"      Data Points: {stats['count']}")
                print()
    else:
        print("⚠️ No impacts found for Spain flood events")
        print("\nℹ️ This could mean:")
        print("  • Impact data hasn't been reported yet")
        print("  • Data is available in different collections")
        print("  • Different date range needed")
        print("  • Different hazard codes required")
    
else:
    print("ℹ️ No Spain flood events found to search for impacts")


📦 Collection: emdat-impacts (7 impacts)
--------------------------------------------------------------------------------

  1. Impact ID: emdat-impact-2024-0839-ESP-total_dam
     Correlation ID: 20241113-ESP-NAT-HYD-FLO-FLO-1-GCDB
     Date: 2024-11-13T00:00:00Z
     Title: Flood (General) in Spain - total_dam

     Impact Detail:
       • Category: total_affected
       • Type: cost
       • Value: 50,000
       • Unit: count
       • Estimate Type: primary

  2. Impact ID: emdat-impact-2024-0839-ESP-total_affected
     Correlation ID: 20241113-ESP-NAT-HYD-FLO-FLO-1-GCDB
     Date: 2024-11-13T00:00:00Z
     Title: Flood (General) in Spain - total_affected

     Impact Detail:
       • Category: people
       • Type: affected_total
       • Value: 500
       • Unit: count
       • Estimate Type: primary

  3. Impact ID: emdat-impact-2024-0839-ESP-no_affected
     Correlation ID: 20241113-ESP-NAT-HYD-FLO-FLO-1-GCDB
     Date: 2024-11-13T00:00:00Z
     Title: Flood (General) in Spain -

In [ ]:
# Create a DataFrame with all impact data
if impacts:
    impact_data = []
    
    for impact in impacts:
        impact_detail = impact.properties.get('monty:impact_detail', {})
        
        # Extract all relevant fields
        impact_record = {
            'Impact ID': impact.id,
            'Collection': impact.collection_id,
            'Correlation ID': impact.properties.get('monty:corr_id', 'N/A'),
            'Date': impact.properties.get('datetime', 'N/A'),
            'Title': impact.properties.get('title', 'N/A'),
            'Category': impact_detail.get('category', 'N/A') if impact_detail else 'N/A',
            'Type': impact_detail.get('type', 'N/A') if impact_detail else 'N/A',
            'Value': impact_detail.get('value', None) if impact_detail else None,
            'Unit': impact_detail.get('unit', 'N/A') if impact_detail else 'N/A',
            'Estimate Type': impact_detail.get('estimate_type', 'N/A') if impact_detail else 'N/A'
        }
        
        impact_data.append(impact_record)
    
    # Create DataFrame
    impacts_df = pd.DataFrame(impact_data)
    
    # Display summary
    print(f"📊 Impact Data Summary")
    print("="*80)
    print(f"Total Impacts: {len(impacts_df)}")
    print(f"\nDataFrame Shape: {impacts_df.shape}")
    print(f"Columns: {list(impacts_df.columns)}\n")
    
    # Display the DataFrame
    print("="*80)
    print("\n📋 All Impacts Table:\n")
    display(impacts_df)
    
    # Summary statistics by category and type
    print("\n" + "="*80)
    print("\n📈 Summary by Category and Type:\n")
    
    summary = impacts_df.groupby(['Category', 'Type']).agg({
        'Value': ['count', 'sum', 'mean', 'min', 'max']
    }).round(0)
    
    # Reset index to make Category and Type regular columns
    summary = summary.reset_index()
    
    # Flatten the multi-level column names
    summary.columns = ['Category', 'Type', 'Count', 'Sum', 'Mean', 'Min', 'Max']
    
    display(summary)
    
else:
    print("⚠️ No impacts available to create DataFrame")

📊 Impact Data Summary
Total Impacts: 46

DataFrame Shape: (46, 10)
Columns: ['Impact ID', 'Collection', 'Correlation ID', 'Date', 'Title', 'Category', 'Type', 'Value', 'Unit', 'Estimate Type']


📋 All Impacts Table:



Impact ID  \
0                                  idmc-idu-impact-28064168623-displaced   
1                    idmc-gidd-impact-28064168623-Internal Displacements   
2                                   emdat-impact-2024-0839-ESP-total_dam   
3                              emdat-impact-2024-0839-ESP-total_affected   
4                                 emdat-impact-2024-0839-ESP-no_affected   
5                    idmc-gidd-impact-27864173831-Internal Displacements   
6                                      idmc-gidd-impact-27864171778-IDPs   
7                    idmc-gidd-impact-27864173829-Internal Displacements   
8                                      idmc-gidd-impact-27864171774-IDPs   
9                                  idmc-idu-impact-27864170550-sheltered   
10                   idmc-gidd-impact-27864173828-Internal Displacements   
11                   idmc-gidd-impact-27864170550-Internal Displacements   
12                   idmc-gidd-impact-27864173827-Internal Displacements   
13                   idmc-gidd-impact-27864173826-Internal Displacements   
14                   idmc-gidd-impact-27864173825-Internal Displacements   
15                   idmc-gidd-impact-27864173822-Internal Displacements   
16                   idmc-gidd-impact-27864173824-Internal Displacements   
17                   idmc-gidd-impact-27864173821-Internal Displacements   
18                                 idmc-idu-impact-27864170560-sheltered   
19                                 idmc-idu-impact-27864170539-sheltered   
20                                 idmc-idu-impact-27864170537-sheltered   
21                   idmc-gidd-impact-27864170539-Internal Displacements   
22  gdacs-hazard-1102983-2-D-bridges destroyed-Spain-Ribarroja del Turia   
23                        gdacs-hazard-1102983-2-B-rescued-Spain-Cullera   
24                        gdacs-hazard-1102983-2-B-displaced-Spain-Letur   
25                      gdacs-hazard-1102983-2-B-displaced-Spain-Alfafar   
26                          gdacs-hazard-1102983-2-A-missing-Spain-Letur   
27                  gdacs-hazard-1102983-2-A-death-Spain-Valencia region   
28                             gdacs-hazard-1102983-2-A-death-Spain-Mira   
29                            gdacs-hazard-1102983-2-A-death-Spain-Letur   
30                        gdacs-hazard-1102983-2-A-death-Spain-Andalusia   
31                        gdacs-hazard-1102983-1-A-death-Spain-Andalusia   
32                gdacs-hazard-1102983-2-A-death-Spain-Castile-La Mancha   
33                gdacs-hazard-1102983-1-A-death-Spain-Castile-La Mancha   
34          gdacs-hazard-1102983-2-B-displaced-Spain-Valencian Community   
35          gdacs-hazard-1102983-1-B-displaced-Spain-Valencian Community   
36              gdacs-hazard-1102983-2-A-death-Spain-Valencian Community   
37              gdacs-hazard-1102983-1-A-death-Spain-Valencian Community   
38     gdacs-hazard-1102983-2-C-houses damaged-Spain-Valencian Community   
39     gdacs-hazard-1102983-1-C-houses damaged-Spain-Valencian Community   
40            gdacs-hazard-1102983-2-A-missing-Spain-Valencian Community   
41            gdacs-hazard-1102983-1-A-missing-Spain-Valencian Community   
42                               emdat-impact-2024-0796-ESP-total_deaths   
43                                  emdat-impact-2024-0796-ESP-total_dam   
44                             emdat-impact-2024-0796-ESP-total_affected   
45                                emdat-impact-2024-0796-ESP-no_affected   

           Collection                       Correlation ID  \
0    idmc-idu-impacts  20241113-ESP-NAT-HYD-FLO-FLO-1-GCDB   
1   idmc-gidd-impacts           20241113-ESP-MH0600-1-GCDB   
2       emdat-impacts  20241113-ESP-NAT-HYD-FLO-FLO-1-GCDB   
3       emdat-impacts  20241113-ESP-NAT-HYD-FLO-FLO-1-GCDB   
4       emdat-impacts  20241113-ESP-NAT-HYD-FLO-FLO-1-GCDB   
5   idmc-gidd-impacts           20241030-ESP-MH0600-1-GCDB   
6   idmc-gidd-impacts           20241030-ESP-MH0600-1-GC



📈 Summary by Category and Type:



Category                Type  Count         Sum       Mean      Min  \
0       buildings             damaged      2      9000.0     4500.0   4500.0   
1       buildings           destroyed      1         1.0        1.0      1.0   
2          people      affected_total      4     73230.0    18308.0    500.0   
3          people            assisted      1        26.0       26.0     26.0   
4          people               death     10       738.0       74.0      1.0   
5          people  displaced_internal     15     12285.0      819.0      0.0   
6          people             missing      3       233.0       78.0      5.0   
7          people           relocated      4      1034.0      258.0     40.0   
8          people   shelter_emergency      4      4293.0     1073.0      8.0   
9  total_affected                cost      2  11050000.0  5525000.0  50000.0   

          Max  
0      4500.0  
1         1.0  
2     36115.0  
3        26.0  
4       232.0  
5      4210.0  
6       114.0  
7       447.0  
8      3701.0  
9  11000000.0

---

## 9. Summary and Key Takeaways

### What We've Accomplished:

In this notebook, we focused on **querying and retrieving disaster data** from the Montandon STAC API, with a detailed case study on Spain flood events (October-November 2024).

#### 1. **API Connection & Collections**
   - Connected to the Montandon STAC API staging environment
   - Explored 30+ collections organized by data source and type (events, hazards, impacts)
   - Understood the structure: GDACS, EMDAT, GLIDE, IDMC, USGS data sources

#### 2. **Basic Query Techniques**
   - Simple collection browsing with `fetch_collection_items()`
   - Direct API requests using HTTP requests
   - Parameter-based filtering with `limit` and collection selection

#### 3. **Advanced CQL2 Filtering**
   - **Temporal filtering**: Date range queries (e.g., October 20 - November 15, 2024)
   - **Spatial filtering**: Bounding box queries (e.g., Europe region)
   - **Property filtering**: Country codes (`monty:country_codes`), hazard codes (`monty:hazard_codes`)
   - **Complex operators**: `a_contains`, `a_overlaps`, `t_intersects`

#### 4. **Event-Hazard-Impact Correlation**
   - Found 4 flood events in Spain using CQL2 filters
   - Retrieved 4 related hazards using correlation IDs (`monty:corr_id`)
   - Discovered 46 impact records across 4 collections:
     - 7 from EMDAT impacts
     - 20 from GDACS impacts
     - 14 from IDMC-GIDD impacts
     - 5 from IDMC-IDU impacts

#### 5. **Data Structure Understanding**
   - Examined `monty:impact_detail` structure with fields: category, type, value, unit, estimate_type
   - Identified impact types: displaced_internal, cost, etc.
   - Mapped impact categories: people, total_affected, buildings, etc.

#### 6. **Data Organization**
   - Converted 46 impact items into a pandas DataFrame
   - Organized data with 10 columns: Impact ID, Collection, Correlation ID, Date, Title, Category, Type, Value, Unit, Estimate Type
   - Created a summary table grouping impacts by Category and Type with counts, sums, means, min/max values

### Key Properties Reference:

| Property | Description | Example |
|----------|-------------|---------|
| `monty:corr_id` | Correlation ID linking events, hazards, impacts | `20241113-ESP-NAT-HYD-FLO-FLO-1-GCDB` |
| `monty:country_codes` | ISO 3166-1 alpha-3 country codes | `['ESP']` for Spain |
| `monty:hazard_codes` | Hazard classification codes | `['nat-hyd-flo-flo', 'FL']` for floods |
| `monty:impact_detail` | Detailed impact metrics (category, type, value, unit, estimate_type) | See impact data |
| `datetime` | Event/hazard/impact timestamp | ISO 8601 format |

### Case Study Results: Spain Floods October-November 2024

**Events Found**: 4
- GDACS Event 1102983, GLIDE Event FL-2024-000199-ESP, EMDAT Event 2024-0796, GDACS Event 1102955

**Hazards Found**: 4 total
- 3 from GDACS hazards
- 1 from GLIDE hazards

**Impacts Found**: 46 total
- Primary impact types: Internal displacements (4,210+ people), economic costs (50,000+), affected population counts
- Data spans November 13-14, 2024 with focus on Andalusia region (Málaga, Granada areas)

### Next Notebook:

The next notebook will cover:
1. **Data Visualization** - Create charts and maps of disaster events
2. **Statistical Analysis** - Analyze patterns and trends in the data
3. **Data Export** - Export results to CSV, GeoJSON, and other formats
4. **Geospatial Analysis** - Perform spatial analysis with GeoPandas
5. **Time Series Analysis** - Analyze temporal patterns of disasters

---

## 🎯 Practice Exercise

Try this on your own:

1. **Search for flood events** in your country or region of interest
2. **Filter by date range** (e.g., last 2 years)
3. **Convert to DataFrame** and analyze the patterns
4. **Find related hazards** using correlation IDs
5. **Export the results** for your records

Use the code cells below to practice:

In [ ]:
# Your practice code here
# Example: Search for floods in your region


---

**End of Notebook 1**

Continue to the next notebook for advanced visualization and spatial analysis!